swap/swap.go

```diff
@@ -267,7 +216,7 @@ func createFactory(factoryAddress common.Address, chainID *big.Int, backend chai
 }
 
 // checkChainID verifies whether we have initialized SWAP before and ensures that we are on the same backendNetworkID if this is the case
```

Comment:
Add a comment, explaining that we pass in a logger, because the global swap object is not yet created when we call this function.

Reply:
Or explain this at the point where you call the function.

Reply:
Added a comment on the logger parameter.
It's a good idea to be explicit with the logger.

Thanks